File: src/applications/serializers/course_paper.py
from rest_framework import serializers 
from applications.model.stages import Stages
from rest_enumfield import EnumField


class GetCoursePapersQuerySerializer(serializers.Serializer):
    order_by = serializers.CharField(required=False, default='created_at')


class CreateCoursePaperDTOSerializer(serializers.Serializer):
    student_id = serializers.UUIDField()
    name = serializers.CharField()


class CoursePaperSerializer(serializers.Serializer):
    id = serializers.UUIDField()
    name = serializers.CharField()
    student_id = serializers.UUIDField()
    current_stage = EnumField(choices=Stages)
    created_at = serializers.DateTimeField()


class CoursePapersListSerializer(serializers.Serializer):
    course_papers = CoursePaperSerializer(many=True)

File: src/web/internal/storage/models/course_paper.go
package models

import (
	"time"
)

type CoursePaper struct {
	ID           string    `json:"id"`
	Name         string    `json:"name"`
	StudentID    string    `json:"student_id"`
	CurrentStage string    `json:"current_stage"`
	CreatedAt    time.Time `json:"created_at"`
}
